### PR TITLE
test(form): add cypress tests

### PR DESCRIPTION
**Pull request by divyajindel**
### Proposed behaviour
Refactor for form component to use new cypress-component-react framework for testing.
I have added test.js and deleted steps.js and feature file.

### Current behaviour

Currently we are using the regular cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

- [x]  Run npx cypress open --component to check if there is newly added test.file
- [x]  Check if the form.test.js file passed
- [x]  Compare all tests were moved/refactored from .feature file -> .test.js file
- [x] Check if all unnecessary files were deleted, including .json file

### CODE REVIEW

**Review by ZhuoyuJin — CHANGES_REQUESTED** [review]

File: cypress/locators/form/locators.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// component preview locators
+export const FORM_COMPONENT = '[data-component="form"]';
+
```
Comment (DlgSHi) [inline cypress/locators/form/locators.js:3]: why do we need this empty line?

File: src/components/form/form.test.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
+/* eslint-disable no-console */
+import React from "react";
+import Form from "../form/form.component";
+import Textbox from "../textbox/textbox.component";
+import Button from "../button/button.component";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+import { useJQueryCssValueAndAssert } from "../../../cypress/support/component-helper/common-steps";
+import {
+  FormPreview,
```
Comment (DlgSHi) [inline src/components/form/form.test.js:9]:
```suggestion
  formPreview,
```

File: cypress/locators/form/index.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { FORM_COMPONENT, FORM_FOOTER_COMPONENT } from "./locators";
+
+// component preview locators
+export const FormPreview = () => cy.get(FORM_COMPONENT);
```
Comment (DlgSHi) [inline cypress/locators/form/index.js:4]:
```suggestion
export const formPreview = () => cy.get(FORM_COMPONENT);
```

File: src/components/form/form.test.js [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+      [3, 335],
+      [5, 367],
+      [7, 407],
+    ])(
+      "should check fieldSpacing as %s for Form component",
+      (fieldSpacing, formHeight) => {
+        CypressMountWithProviders(
+          <FormComponent fieldSpacing={fieldSpacing} />
+        );
+        FormPreview().then(($el) => {
```
Comment (DlgSHi) [inline src/components/form/form.test.js:84]:
```suggestion
        formPreview().then(($el) => {
```

File: src/components/form/form.test.js [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+        .and("be.visible");
+    });
+
+    it.each([true, false])(
+      "should check noValidate set as %s for Form component",
+      (noValidate) => {
+        CypressMountWithProviders(<FormComponent noValidate={noValidate} />);
+
+        if (noValidate === true) {
+          FormPreview().should("have.attr", "noValidate");
```
Comment (DlgSHi) [inline src/components/form/form.test.js:112]:
```suggestion
          formPreview().should("have.attr", "noValidate");
```

File: src/components/form/form.test.js [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+        } else {
+          FormPreview().should("not.have.attr", "noValidate");
```
Comment (DlgSHi) [inline src/components/form/form.test.js:114]:
```suggestion
          formPreview().should("not.have.attr", "noValidate");
```

File: src/components/form/form.test.js [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+        }
+      }
+    );
+
+    describe("check events for form component", () => {
+      let callback;
+      beforeEach(() => {
+        callback = cy.stub();
+      });
+      it("should call onSubmit callback when a click event is triggered", () => {
```
Comment (DlgSHi) [inline src/components/form/form.test.js:124]: if we have only 1 test for `onSubmit` why we need to create a separate `describe` block?
Even if we've created it why we need to put all other tests here?

File: src/components/form/form.test.js [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+        CypressMountWithProviders(<FormComponent onSubmit={callback} />);
+
+        FormPreview()
```
Comment (DlgSHi) [inline src/components/form/form.test.js:127]:
```suggestion
        formPreview()
```

File: src/components/form/form.test.js [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+            expect(callback);
+          });
+      });
+    });
+
+    it("should check rightSideButtons for Form component", () => {
+      CypressMountWithProviders(
+        <FormComponent
+          rightSideButtons={
+            <Button onClick={() => console.log("other")} ml={2}>
```
Comment (DlgSHi) [inline src/components/form/form.test.js:140]:
```suggestion
            <Button onClick={() => {}} ml={2}>
```

File: src/components/form/form.test.js [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+          .and("have.text", `${warningCount} warnings`)
+          .and("be.visible");
+      }
+    );
+
+    it.each(["30px", "100px", "200px"])(
+      "should check height for Form component",
+      (height) => {
+        CypressMountWithProviders(<FormComponent height={height} />);
+        FormPreview().should("have.css", "height", height);
```
Comment (DlgSHi) [inline src/components/form/form.test.js:195]:
```suggestion
        formPreview().should("have.css", "height", height);
```

**Review by DlgSHi — CHANGES_REQUESTED** [review]

File: src/components/form/form.test.js [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+          useJQueryCssValueAndAssert($el, "height", formHeight);
+        });
+      }
+    );
+
+    it("should check leftSideButtons for Form component", () => {
+      CypressMountWithProviders(
+        <FormComponent
+          leftSideButtons={
+            <Button onClick={() => console.log("cancel")} ml={2}>
```
Comment (DlgSHi) [inline src/components/form/form.test.js:96]: this is still not fixed
```suggestion
            <Button onClick={() => {}} ml={2}>
```

File: src/components/form/form.test.js [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+        .should("have.text", "Cancel")
+        .and("be.visible");
+    });
+
+    it.each([true, false])(
+      "should check noValidate set as %s for Form component",
+      (noValidate) => {
+        CypressMountWithProviders(<FormComponent noValidate={noValidate} />);
+
+        if (noValidate === true) {
```
Comment (DlgSHi) [inline src/components/form/form.test.js:113]:
```suggestion
        if (noValidate) {
```

File: src/components/form/form.test.js [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+import { useJQueryCssValueAndAssert } from "../../../cypress/support/component-helper/common-steps";
+import { getElement, getComponent } from "../../../cypress/locators/index";
+import {
+  formPreview,
+  formFooterComponent,
+  formButtonComponent,
+} from "../../../cypress/locators/form/index";
+
+let callback;
```
Comment (DlgSHi) [inline src/components/form/form.test.js:14]: we can delete this and just use only in test with `callback`

File: src/components/form/form.test.js [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+        if (noValidate === true) {
+          formPreview().should("have.attr", "noValidate");
+        } else {
+          formPreview().should("not.have.attr", "noValidate");
+        }
+      }
+    );
+
+    it("should call onSubmit callback when a click event is triggered", () => {
+      CypressMountWithProviders(<FormComponent onSubmit={callback} />);
```
Comment (DlgSHi) [inline src/components/form/form.test.js:122]:
```suggestion
      const callback = cy.stub();

      CypressMountWithProviders(<FormComponent onSubmit={callback} />);
```

**Review by DlgSHi — CHANGES_REQUESTED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**